integration-tests/smoke/ccip/ccip_rmn_test.go

```diff
@@ -448,9 +404,241 @@ type rmnTestCase struct {
+	// At the moment, it does not support waitForExec=true since only commit plugin has cursing checks.
+	cursedSourceChainIdxs []int
+	// globalCurse marks every chain as cursed by setting the global curse subject on each rmnRemote
+	globalCurse        bool
+	waitForExec        bool
+	homeChainConfig    homeChainConfig
+	remoteChainsConfig []remoteChainConfig
+	rmnNodes           []rmnNode
+	messagesToSend     []messageToSend
+
+	// populated fields after environment setup
+	pf testCasePopulatedFields
+}
+
+type testCasePopulatedFields struct {
+	chainSelectors      []uint64
+	rmnHomeNodes        []rmn_home.RMNHomeNode
+	rmnRemoteSigners    []rmn_remote.RMNRemoteSigner
+	rmnHomeSourceChains []rmn_home.RMNHomeSourceChain
+	cursedSourceChains  mapset.Set[uint64]
+}
+
+func (tc *rmnTestCase) populateFields(t *testing.T, envWithRMN changeset.DeployedEnv, rmnCluster devenv.RMNCluster) {
+	require.GreaterOrEqual(t, len(envWithRMN.Env.Chains), 2, "test assumes at least two chains")
+	for _, chain := range envWithRMN.Env.Chains {
+		tc.pf.chainSelectors = append(tc.pf.chainSelectors, chain.Selector)
+	}
+
+	for _, rmnNodeInfo := range tc.rmnNodes {
+		rmn := rmnCluster.Nodes["rmn_"+strconv.Itoa(rmnNodeInfo.id)]
+
+		var offchainPublicKey [32]byte
+		copy(offchainPublicKey[:], rmn.RMN.OffchainPublicKey)
+
+		tc.pf.rmnHomeNodes = append(tc.pf.rmnHomeNodes, rmn_home.RMNHomeNode{
+			PeerId:            rmn.Proxy.PeerID,
+			OffchainPublicKey: offchainPublicKey,
+		})
+
+		if rmnNodeInfo.isSigner {
+			if rmnNodeInfo.id < 0 {
+				t.Fatalf("node id is negative: %d", rmnNodeInfo.id)
+			}
+			tc.pf.rmnRemoteSigners = append(tc.pf.rmnRemoteSigners, rmn_remote.RMNRemoteSigner{
+				OnchainPublicKey: rmn.RMN.EVMOnchainPublicKey,
+				NodeIndex:        uint64(rmnNodeInfo.id),
+			})
+		}
+	}
+
+	for remoteChainIdx, remoteF := range tc.homeChainConfig.f {
+		if remoteF < 0 {
+			t.Fatalf("negative remote F: %d", remoteF)
+		}
+		// configure remote chain details on the home contract
+		tc.pf.rmnHomeSourceChains = append(tc.pf.rmnHomeSourceChains, rmn_home.RMNHomeSourceChain{
+			ChainSelector:       tc.pf.chainSelectors[remoteChainIdx],
+			F:                   uint64(remoteF),
+			ObserverNodesBitmap: createObserverNodesBitmap(tc.pf.chainSelectors[remoteChainIdx], tc.rmnNodes, tc.pf.chainSelectors),
+		})
+	}
+
+	tc.pf.cursedSourceChains = mapset.NewSet[uint64]()
+	for _, chainIdx := range tc.cursedSourceChainIdxs {
+		tc.pf.cursedSourceChains.Add(tc.pf.chainSelectors[chainIdx])
+	}
+}
+
+func (tc rmnTestCase) validate() error {
+	if len(tc.cursedSourceChainIdxs) > 0 {
+		if tc.waitForExec {
+			return errors.New("cursedSourceChainIdxs is set but waitForExec is true which is not supported")
+		}
+		if tc.passIfNoCommitAfter == 0 {
+			return errors.New("cursedSourceChainIdxs is set but passIfNoCommitAfter is not set")
+		}
+	}
+
+	if tc.globalCurse {
```

Comment:
I'm kinda confused by these settings - if a source chain is cursed, the chain itself is cursed with GLOBAL_CURSE_SUBJECT. What do you mean by global curse here?